File: sample-chat-java/app/src/main/java/com/quickblox/sample/chat/java/ui/adapter/CheckboxUsersAdapter.java
package com.quickblox.sample.chat.java.ui.adapter;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import com.quickblox.sample.chat.java.R;
import com.quickblox.sample.chat.java.utils.ResourceUtils;
import com.quickblox.users.model.QBUser;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class CheckboxUsersAdapter extends UsersAdapter {

    private List<Integer> initiallySelectedUsers;
    private Set<QBUser> selectedUsers;

    public CheckboxUsersAdapter(Context context, List<QBUser> users) {
        super(context, users);
        selectedUsers = new HashSet<>();
        initiallySelectedUsers = new ArrayList<>();
    }

    public Set<QBUser> getSelectedUsers() {
        return selectedUsers;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view = super.getView(position, convertView, parent);
        final QBUser user = getItem(position);
        final ViewHolder holder = (ViewHolder) view.getTag();

        holder.userCheckBox.setVisibility(View.VISIBLE);
        boolean containsUser = selectedUsers.contains(user);
        holder.userCheckBox.setChecked(containsUser);

        if (containsUser) {
            holder.rootLayout.setBackgroundColor(ResourceUtils.getColor(R.color.selected_list_item_color));
        } else {
            holder.rootLayout.setBackgroundColor(ResourceUtils.getColor(android.R.color.transparent));
        }

        return view;
    }

    public void onItemClicked(int position, View convertView, ViewGroup parent) {
        QBUser user = getItem(position);
        ViewHolder holder = (UsersAdapter.ViewHolder) convertView.getTag();

        if (!isAvailableForSelection(user)) {
            return;
        }

        holder.userCheckBox.setChecked(!holder.userCheckBox.isChecked());
        if (holder.userCheckBox.isChecked()) {
            selectedUsers.add(user);
        } else {
            selectedUsers.remove(user);
        }
        notifyDataSetChanged();
    }

    @Override
    public QBUser getItem(int position) {
        return userList.get(position);
    }

    @Override
    protected boolean isAvailableForSelection(QBUser user) {
        return super.isAvailableForSelection(user) && !initiallySelectedUsers.contains(user.getId());
    }
}

File: sample-chat-kotlin/app/src/main/java/com/quickblox/sample/chat/kotlin/ui/adapter/CheckboxUsersAdapter.kt
package com.quickblox.sample.chat.kotlin.ui.adapter

import android.content.Context
import android.view.View
import android.view.ViewGroup
import com.quickblox.sample.chat.kotlin.R
import com.quickblox.users.model.QBUser


class CheckboxUsersAdapter(context: Context, users: List<QBUser>) : UsersAdapter(context, users as MutableList<QBUser>) {

    private val initiallySelectedUsers: MutableList<Int> = ArrayList()
    private val _selectedUsers: MutableSet<QBUser> = HashSet()
    val selectedUsers: ArrayList<QBUser>
        get() = ArrayList<QBUser>(_selectedUsers)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view = super.getView(position, convertView, parent)
        val user = getItem(position)
        val holder = view.tag as UsersAdapter.ViewHolder

        holder.userCheckBox.visibility = View.VISIBLE
        val containsUser = _selectedUsers.contains(user)
        holder.userCheckBox.isChecked = containsUser

        if (containsUser) {
            holder.rootLayout.setBackgroundColor(context.resources.getColor(R.color.selected_list_item_color))
        } else {
            holder.rootLayout.setBackgroundColor(context.resources.getColor(android.R.color.transparent))
        }

        return view
    }

    fun onItemClicked(position: Int, convertView: View?, parent: ViewGroup) {
        val user = getItem(position)
        val holder = convertView?.tag as UsersAdapter.ViewHolder

        if (!isAvailableForSelection(user)) {
            return
        }

        holder.userCheckBox.isChecked = !holder.userCheckBox.isChecked
        if (holder.userCheckBox.isChecked) {
            _selectedUsers.add(user)
        } else {
            _selectedUsers.remove(user)
        }
        notifyDataSetChanged()
    }

    override fun getItem(position: Int): QBUser {
        return userList[position]
    }

    override fun isAvailableForSelection(user: QBUser): Boolean {
        return super.isAvailableForSelection(user) && !initiallySelectedUsers.contains(user.id)
    }
}